control-plane-operator/controllers/hostedcontrolplane/registryoperator/testdata/zz_fixture_TestReconcileDeployment.yaml

```
@@ -0,0 +1,201 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  creationTimestamp: null
+  labels:
+    hypershift.openshift.io/managed-by: control-plane-operator
+  name: cluster-image-registry-operator
+  namespace: test-namespace
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      name: cluster-image-registry-operator
+  strategy:
+    type: Recreate
+  template:
+    metadata:
+      annotations:
+        hypershift.openshift.io/release-image: quay.io/ocp-dev/test-release-image:latest
+      creationTimestamp: null
+      labels:
+        hypershift.openshift.io/hosted-control-plane: test-namespace
+        name: cluster-image-registry-operator
+    spec:
+      affinity:
+        nodeAffinity:
+          preferredDuringSchedulingIgnoredDuringExecution:
+          - preference:
+              matchExpressions:
+              - key: hypershift.openshift.io/control-plane
+                operator: In
+                values:
+                - "true"
+            weight: 50
+          - preference:
+              matchExpressions:
+              - key: hypershift.openshift.io/cluster
+                operator: In
+                values:
+                - test-namespace
+            weight: 100
+        podAffinity:
+          preferredDuringSchedulingIgnoredDuringExecution:
+          - podAffinityTerm:
+              labelSelector:
+                matchLabels:
+                  hypershift.openshift.io/hosted-control-plane: test-namespace
+              topologyKey: kubernetes.io/hostname
+            weight: 100
+      automountServiceAccountToken: false
+      containers:
+      - args:
+        - -c
+        - |
+          #!/bin/bash
+          set -euo pipefail
+
+          while true; do
+             if [[ -f /var/run/secrets/kubernetes.io/serviceaccount/token ]]; then
+                break
+             fi
+             echo "Waiting for client token"
+             sleep 2
+          done
+
+          echo "openshift-image-registry" > "/var/run/secrets/kubernetes.io/serviceaccount/namespace"
+          cp "/etc/certificate/ca/ca.crt" "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
+          export KUBERNETES_SERVICE_HOST=kube-apiserver
+          export KUBERNETES_SERVICE_PORT=$KUBE_APISERVER_SERVICE_PORT
+          /usr/bin/cluster-image-registry-operator \
```

Nit to stop running the shell:
```suggestion
          exec /usr/bin/cluster-image-registry-operator \
```